axelrod/strategies/adaptive.py

```diff
@@ -0,0 +1,50 @@
+from axelrod import Actions, Player, Game
+
+C, D = Actions.C, Actions.D
+
+
+class Adaptive(Player):
+    """Start with a specific sequence of C and D, then play the strategy that
+    has worked best, recalculated each turn."""
+
+    name = 'Adaptive'
+    classifier = {
+        'memory_depth': float('inf'),  # Long memory
+        'stochastic': False,
+        'makes_use_of': set(),
```

Comment:
This should include the game.

You'll then need to update the [classification of strategies documentation](https://github.com/Axelrod-Python/Axelrod/blob/master/docs/tutorials/advanced/classification_of_strategies.rst).